temporian/implementation/numpy/operators/tests/lag_test.py

```diff
@@ -72,11 +77,73 @@ def test_correct_lag(self) -> None:
             event=event,
         )
 
-        lag_implementation = lag.LagNumpyImplementation(operator)
+        lag_implementation = LagNumpyImplementation(operator)
         operator_output = lag_implementation(event=numpy_input_event)
 
         self.assertTrue(numpy_output_event == operator_output["event"])
 
+    def test_correct_multiple_lags(self) -> None:
+        """Test correct lag operator with duration list."""
+        numpy_input_event = NumpyEvent.from_dataframe(
+            pd.DataFrame(
+                [
+                    [0, 1.0, 10.0],
+                    [0, 2.0, 11.0],
+                    [0, 3.0, 12.0],
+                    [0, 4.0, 13.0],
+                    [0, 5.0, 14.0],
+                    [0, 6.0, 15.0],
+                    [0, 7.0, 16.0],
+                ],
+                columns=["store_id", "timestamp", "sales"],
+            ),
+            index_names=["store_id"],
+        )
+
+        expected_numpy_output_event = NumpyEvent.from_dataframe(
+            pd.DataFrame(
+                [
+                    [0, 1.0, 10.0, np.nan, np.nan],
+                    [0, 2.0, 11.0, 10.0, np.nan],
+                    [0, 3.0, 12.0, 11.0, 10.0],
+                    [0, 4.0, 13.0, 12.0, 11.0],
+                    [0, 5.0, 14.0, 13.0, 12.0],
+                    [0, 6.0, 15.0, 14.0, 13.0],
+                    [0, 7.0, 16.0, 15.0, 14.0],
+                ],
+                columns=[
+                    "store_id",
+                    "timestamp",
+                    "sales",
+                    "lag[1s]_sales",
+                    "lag[2s]_sales",
+                ],
+            ),
+            index_names=["store_id"],
+        )
+
+        event = numpy_input_event.schema()
+
+        # lag multiple durations
+        lags = lag(event=event, duration=[1, 2])
+
+        # assign multiple lags to output event
+        output_event = event
+        for lagged_event in lags:
+            output_event = assign(output_event, lagged_event)
```

Comment:
Can you test directly the output of the lag/leak operator (instead of using `assign`)?